regression/cbmc/gcc_vector3/main.c

```diff
@@ -31,7 +31,12 @@ int main()
   assert(res.members[1] == 5);
   assert(res.members[2] == 3);
   assert(res.members[3] == 6);
-#elif defined(__clang__)
+#endif
+}
+
+void test_shufflevector(void)
+{
+#if defined(__GNUC__) && (defined(__clang__) || __GNUC__ >= 12)
```

Comment:
This seems wrong, or at least hard to parse. Why would you check the GNUC version when using clang?

Reply:
This is to restrict the GCC version: shufflevector is supported in all versions of clang, but for GCC it's only supported with version 12 or later. I can throw in an explicit `!defined(__clang__)` if that makes parsing easier?

Reply:
How about
`#if defined(__clang__) || (defined(__clang__) && __GNUC__ >= 12)`

Reply:
Done (except replacing the second use of `__clang__` by `__GNUC__`).